coremltools/test/ml_program/test_utils.py

```diff
@@ -892,3 +892,188 @@ def test_10_duplicated_model(self):
                 == 10
             )
         shutil.rmtree(saved_package_path)
```

Comment:
Nice tests! 

One of the issues that the script in the SD repo had, was that even if the original model had FP16 types for I/O (which is possible for ios17 target and above), the chunked model would have casts and FP32 dtypes. Looks like you have addressed that issue in this util which is great. 
Can you add a test to check for that?

Reply:
Done!